dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/cdi/BaggageProxy.java

```diff
@@ -19,47 +19,91 @@
 
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+
 /**
  * This proxy class redirects method calls to Baggage.current(), by doing so it allows people to use @Inject Baggage and get an object which will not become obsolete.
  */
 public class BaggageProxy implements Baggage {
 
+    private static final TraceComponent tc = Tr.register(BaggageProxy.class);
+
     @Override
     public int size() {
-        return Baggage.current().size();
+        try {
+            return Baggage.current().size();
+        } catch (Exception e) {
+            Tr.error(tc, Tr.formatMessage(tc, "CEMOT5101.telemetry.error", e));
+            return -1;
+        }
     }
 
     @Override
     public void forEach(BiConsumer<? super String, ? super BaggageEntry> consumer) {
-        Baggage.current().forEach(consumer);
+        try {
+            Baggage.current().forEach(consumer);
+        } catch (Exception e) {
+            Tr.error(tc, Tr.formatMessage(tc, "CEMOT5101.telemetry.error", e));
+        }
     }
 
     @Override
     public Map<String, BaggageEntry> asMap() {
-        return Baggage.current().asMap();
+        try {
+            return Baggage.current().asMap();
+        } catch (Exception e) {
+            Tr.error(tc, Tr.formatMessage(tc, "CEMOT5101.telemetry.error", e));
+            return null;
+        }
     }
 
     @Override
     public String getEntryValue(String entryKey) {
-        return Baggage.current().getEntryValue(entryKey);
+        try {
+            return Baggage.current().getEntryValue(entryKey);
+        } catch (Exception e) {
+            Tr.error(tc, Tr.formatMessage(tc, "CEMOT5101.telemetry.error", e));
+            return null;
+        }
     }
 
     @Override
     public BaggageBuilder toBuilder() {
-        return Baggage.current().toBuilder();
+        try {
+            return Baggage.current().toBuilder();
+        } catch (Exception e) {
+            Tr.error(tc, Tr.formatMessage(tc, "CEMOT5101.telemetry.error", e));
+            return null;
+        }
     }
     
     @Override
     public boolean equals(Object obj) {
-        return Baggage.current().equals(obj);
+        try {
+            return Baggage.current().equals(obj);
+        } catch (Exception e) {
+            Tr.error(tc, Tr.formatMessage(tc, "CEMOT5101.telemetry.error", e));
+            return false;
+        }
     }
 
     public int hashCode() {
-        return Baggage.current().hashCode();
+        try {
+            return Baggage.current().hashCode();
+        } catch (Exception e) {
+            Tr.error(tc, Tr.formatMessage(tc, "CEMOT5101.telemetry.error", e));
+            return -1;
+        }
     }
 
     @Override
     public String toString() {
-        return Baggage.current().toString();
+        try {
+            return Baggage.current().toString();
+        } catch (Exception e) {
+            Tr.error(tc, Tr.formatMessage(tc, "CEMOT5101.telemetry.error", e));
+            return null;
```

Comment:
`null` is likely to cause further problems. It's probably preferable to return an empty string here.